src/GraphBLAS-sharp.Backend/Matrix/COO/Matrix.fs

```diff
@@ -216,3 +236,106 @@ module Matrix =
               Columns = copy queue allocationMode matrix.Columns
               Values = copyData queue allocationMode matrix.Values }
             |> transposeInPlace queue
+
+    /// <summary>
+    /// Builds a bitmap. Maps non-zero elements of the left matrix
+    /// to 1 if the right matrix has non zero element under the same row and column pair, otherwise 0.
+    /// </summary>
+    /// <param name="clContext">OpenCL context.</param>
+    /// <param name="workGroupSize">Should be a power of 2 and greater than 1.</param>
+    let findKeysIntersection (clContext: ClContext) workGroupSize =
+        Intersect.findKeysIntersection clContext workGroupSize
+
+    /// <summary>
+    /// Merges two disjoint matrices of the same size.
+    /// </summary>
+    /// <remarks>
+    /// Matrices should have the same number of rows and columns. <br/>
+    /// Matrices should not have non zero values with the same index.
+    /// </remarks>
+    /// <param name="clContext">OpenCL context.</param>
+    /// <param name="workGroupSize">Should be a power of 2 and greater than 1.</param>
+    let mergeDisjoint (clContext: ClContext) workGroupSize =
```

Comment:
We can implement it using map2, but, I guess, it will be less efficient, because inside map2 dense matrix is allocated first. MergeDisjoint allocates matrix of the required size without the need to truncate it later.

Reply:
Ok. Looks reasonable.